### DISCUSSION TITLE
Using `sgpt` to cat or bat a file to run GPT against

### DISCUSSION THREAD

**Discussion by o6uoq** [tbckr/sgpt#265] [Q&A]
**As an**  SGPT user
**I want** to be able to [cat](https://phoenixnap.com/kb/linux-cat-command) or [bat](https://github.com/sharkdp/bat) files within shell
**So that** I am able to process analytics and all the magic of GPT via inputting a file from disk/shell into `sgpt` command

**Comment by tbckr** [comment]
Hey @o6uoq,

good question! You can pass in data from `cat` / `bat` with the following command:

`cat myfile.txt | sgpt stdin "this is my additional prompt to say what i should do with the piped in file data"`

You only have to provide `stdin` as a persona / modifier. With the second argument you can provide an additional prompt.

If you have any other question, feel free to ask!
